mesonbuild/interpreter.py

```diff
@@ -3983,6 +3983,9 @@ def _func_custom_target_impl(self, node, args, kwargs):
                 mlog.warning('''Custom target input \'%s\' can\'t be converted to File object(s).
 This will become a hard error in the future.''' % kwargs['input'], location=self.current_node)
         kwargs['env'] = self.unpack_env_kwarg(kwargs)
+        if 'command' in kwargs and isinstance(kwargs['command'], list) and kwargs['command']:
```

Comment:
I think you should do `command = extract_as_list(kwargs, 'command')` because you also want to replace the command if it's a single string.